src/app/ui/color_wheel.cpp

```diff
@@ -84,6 +84,22 @@ app::Color ColorWheel::getColorInClientPos(const gfx::Point& pos)
   int v = (pos.y - (m_wheelBounds.y+m_wheelBounds.h/2));
   double d = std::sqrt(u*u + v*v);
 
+  if (m_colorModel == ColorModel::NORMAL_MAP) {
+    float x = float(u) / float(m_wheelBounds.w / 2);
+    float y = -float(v) / float(m_wheelBounds.h / 2);
+    float z = std::sqrt(1 - x*x - y*y);
+    if (z <= 1.f) {
+      return app::Color::fromRgb(
+        int(std::round((x + 1) * 127.5)),
+        int(std::round((y + 1) * 127.5)),
+        int(std::round((z + 1) * 127.5))
+      );
+    }
+    else {
+      return app::Color::fromRgb(128, 128, 255);
+    }
+  }
+
   // Pick from the wheel
   if (d < m_wheelRadius+2*guiscale()) {
     double a = std::atan2(-v, u);
```

Comment:
Inside this condition we should be calculating the color map color depending on `d` and `a`.